### PR TITLE
AAE-12578 added fn to save col width with preferences

### PR DESCRIPTION
**Pull request by tomaszhanaj**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://alfresco.atlassian.net/browse/AAE-12578

**What is the new behaviour?**
When user changes the width of a columns, width value is saved with preferences.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/process-services-cloud/src/lib/task/task-list/models/tasks-cloud-preferences.ts [hunk 1]
```diff
@@ -18,5 +18,6 @@
 // eslint-disable-next-line no-shadow
 export enum TasksListCloudPreferences {
     columnOrder = 'tasks-list-cloud-columns-order',
-    columnsVisibility = 'tasks-list-cloud-columns-visibility'
+    columnsVisibility = 'tasks-list-cloud-columns-visibility',
+    columnsWidths = 'tasks-list-cloud-columns-widths',
```
Comment (ehsan-2019) [inline lib/process-services-cloud/src/lib/task/task-list/models/tasks-cloud-preferences.ts:22]: you need to remove the comma ...

File: lib/process-services-cloud/src/lib/task/task-list/components/base-task-list-cloud.component.ts [hunk 2]
```diff
@@ -292,6 +298,23 @@ export abstract class BaseTaskListCloudComponent<T = unknown> extends DataTableS
         this.reload();
     }
 
+    onColumnsWidthChanged(columns: DataColumn[]): void {
```
Comment (ehsan-2019) [inline lib/process-services-cloud/src/lib/task/task-list/components/base-task-list-cloud.component.ts:301]: Please fix the format of this method.

**Review by ehsan-2019 — CHANGES_REQUESTED** [review]

File: lib/process-services-cloud/src/lib/task/task-list/models/tasks-cloud-preferences.ts (hunk 1 above)
Comment (tomaszhanaj) [reply lib/process-services-cloud/src/lib/task/task-list/models/tasks-cloud-preferences.ts:22]: fixed

File: lib/process-services-cloud/src/lib/task/task-list/components/base-task-list-cloud.component.ts (hunk 2 above)
Comment (tomaszhanaj) [reply lib/process-services-cloud/src/lib/task/task-list/components/base-task-list-cloud.component.ts:301]: fixed

**Review by ehsan-2019 — APPROVED** [review]

**Comment by ehsan-2019** [comment]
Please do the same for the process list

**Review by ehsan-2019 — CHANGES_REQUESTED** [review]
Please do the same for the process list

**Review by ehsan-2019 — APPROVED** [review]

**Pull request merged**